lib/controllers/users.js

```diff
@@ -0,0 +1,57 @@
+const { Router } = require('express');
+const UserService = require('../services/UserService');
+const { User } = require('../models/User');
+const authenticate = require('../middleware/authenticate');
+const authorize = require('../middleware/authorize');
+
+const ONE_DAY_IN_MS = 1000 * 60 * 60 * 24;
+
+module.exports = Router()
+  .post('/', async (req, res, next) => {
+    try {
+      const user = await UserService.create(req.body);
+      res.json(user);
+    } catch (error) {
+      next(error);
+    }
+  })
+  .post('/sessions', async (req, res, next) => {
+    try {
+      const token = await UserService.signIn(req.body);
+      res
+        .cookie(process.env.COOKIE_NAME, token, {
+          httpOnly: true,
+          maxAge: ONE_DAY_IN_MS,
+        })
+        .json({ message: 'WELCOME USER, TO YOUR DOOOM!' });
+    } catch (e) {
+      next(e);
+    }
+  })
+
+  .get('/secrets', [authenticate, authorize], async (req, res, next) => {
```

Comment:
you don't need to use the authorize middleware here. we want secrets to be available to all users but not nonusers. authentication is "you are who you say you are" e.g. you are proving you have a ticket to the concert so you can go into the venue. authorization is "you are qualified to be here" e.g. you have a VIP pass to get into the VIP section at the concert. in this app, all users have access to the secrets — the secrets are the band in the venue.